db/view/view.cc

```diff
@@ -332,14 +378,33 @@ static query::partition_slice make_partition_slice(const schema& s) {
             std::move(opts));
 }
 
+view_builder::build_step& view_builder::get_or_create_build_step(utils::UUID base_id) {
+    auto it = _base_to_build_step.find(base_id);
+    if (it == _base_to_build_step.end()) {
+        auto base = _db.find_column_family(base_id).shared_from_this();
+        auto p = _base_to_build_step.emplace(base_id, build_step{base, make_partition_slice(*base->schema())});
+        // Iterators could have been invalidated if there was rehashing, so just reset the cursor.
+        _current_step = p.first;
+        it = p.first;
+    }
+    return it->second;
+}
+
 bool matches_view_filter(const schema& base, const view_info& view, const partition_key& key, const clustering_row& update, gc_clock::time_point now) {
+    std::vector<bytes> exploded_pk = key.explode();
+    std::vector<bytes> exploded_ck = update.key().explode();
+    auto slice = make_partition_slice(base);
+
+    data_query_result_builder builder(base, slice, query::result_options::only_result(), query::result_memory_accounter{10*1024*1024});
+    builder.consume_new_partition(dht::decorate_key(base, key));
+    builder.consume(clustering_row(base, update), row_tombstone{}, update.is_live(base, tombstone{}, now));
```

Comment:
Yes, so we recreate the builder each time. It's inefficient but nothing like the other inefficiencies here, so we can wait with improving it.

Reply:
I see you don't pass the static row. Are views prohibited from filtering on the static row? Reasonable, but let's make sure.

Reply:
This function's signature didn't change and it does not pass any static row, so I think it's discarded earlier in the process. I'll try to verify.

Reply:
Verified in cqlsh. An attempt to restrict a static column in a view filter results in:
```
Non-primary key columns cannot be restricted in the SELECT statement used for materialized view v creation (got restrictions on: s)
```